Generics.md

```diff
@@ -1,164 +1,386 @@
-I’m hoping we can find a fairly minimal design for generics
-that is easy for programmers to learn and supportable with modest effort in wesl tools.
-
-To ease implementation effort, I imagine we’ll want to avoid type inference
-or type constraints on generics. But without type inference, 
-specifying generic types at every call site to a generic function gets verbose and tedious. To avoid that verbosity, let’s allow generic variables on import statements (glslify and wgsl-linker did this too).
-
-I thought we might start by allowing generics only on functions.
-We’ll want a design that’s extensible to more features (e.g. generic structs)
-of course, 
-but we can start with a minimal implementation and add features as they prove necessary.
-
-## Summary
-
-* angle bracket syntax for generic variable declaration, and generic value specification.
-* declare generic variables on function declarations, e.g.: `fn foo<E>(arg: E) -> E { let e:E = arg; return e; }`
-* within an fn with a generic declaration,
-  generic variables names can be used in place of a wgsl type in both the fn declaration and fn body,
-  or in a function call expression inside the fn body.
-  The generic variable text will be replaced by the generic value text during linking.
-  So if `E` is `f32`  the linked wgsl for foo would be: `fn foo(arg: f32) { let e:f32 = arg; return e; }`.
-* Note that a linker will generate multiple copies of fn foo() in wgsl,
-  one for each unique set of generic arguments. So each fn will have a unique name.
-* generic variable values are supplied on import statements or call statements.
-
-      * import with a generic:
-        ```
-        import util/foo<f32> as foo32;
-        main() { foo32(1.0); }
-        ```
-      * or, call with a generic:
-        ```
-        foo<f32>(1.0);
-        ```
-
-* generic values supplied with imports are single world tokens (typically wgsl type names or function names), 
-  or generic variables declared on that function.
+# Generic Declarations and Generic Modules
 
-## Examples
+I would like to limit as much as possible the amount of additional syntax and semantics in our extensions.
+After all, we are writing an extension, not a different language!
+It should look and feel just like the original stuff and extensions should be straight forward, self-explanatory even.
+
+## The proposal
+
+It builds on 3 observations
+
+* WGSL *already* has support for parametric types and functions, they are called [type-generators](https://www.w3.org/TR/WGSL/#type-generator) and [function overloads](https://www.w3.org/TR/WGSL/#builtin-functions). It even has some syntax for them.
+  The issue is, they are not allowed to be declared in user code. This proposal changes this.
+* A *module* is a collection of functions, structs and associated types that share a semantic. Turns out, a WGSL file is just that. This proposal introduces *implicit modules* as files.
```

Comment:
The [glossary](https://github.com/wgsl-tooling-wg/wesl-spec/blob/main/GLOSSARY.md) defined Module as a single WESL file. Are implicit modules something different? 
(guessing it's just distinguishing vs the discussions of explicit namespaces within a file.)